pkg/ingester/ingester.go

```diff
@@ -138,6 +139,32 @@ type LabelValueFilterer interface {
 	Filter(ctx context.Context, labelName string, labelValues []string) ([]string, error)
 }
 
+type Wrapper interface {
+	Wrap(wrapped Ingesterer) Ingesterer
+}
+
+// ChunkStore is the interface we need to store chunks.
+type ChunkStore interface {
+	Put(ctx context.Context, chunks []chunk.Chunk) error
+	SelectLogs(ctx context.Context, req logql.SelectLogParams) (iter.EntryIterator, error)
+	SelectSamples(ctx context.Context, req logql.SelectSampleParams) (iter.SampleIterator, error)
+	GetChunkRefs(ctx context.Context, userID string, from, through model.Time, matchers ...*labels.Matcher) ([][]chunk.Chunk, []*chunk.Fetcher, error)
+	GetSchemaConfigs() []chunk.PeriodConfig
+}
+
+// Ingesterer is an interface for the Ingester
+type Ingesterer interface {
```

Comment:
Can we name this `Interface` instead, so it can be referenced as `ingester.Interface` externally? `Ingester-er` makes me squirm :(

Reply:
Yeah, that was not an ideal name. It's changed now.